src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -91,6 +92,21 @@ public class SocketOutput : ISocketOutput
                 if (buffer.Count > 0)
                 {
                     var tail = ProducingStart();
+                    if (tail.IsDefault)
+                    {
+                        if (cancellationToken.CanBeCanceled)
+                        {
+                            return TaskUtilities.GetCancelledTask(
+                                cancellationToken.IsCancellationRequested ?
+                                cancellationToken :
+                                new CancellationToken(true));
```

Comment:
Might be easier to ask the questions with test behaviour; should this be a cancelled or completed Task? 

https://github.com/benaadams/KestrelHttpServer/blob/socket-error/test/Microsoft.AspNet.Server.KestrelTests/SocketOutputTests.cs#L298-L305